### PR TITLE
Multi head attention support

### PR DESCRIPTION
**Pull request by elad-c**
Support MultiHeadAttention layer in Tensorflow framework:

substitute the layer with an implementation of Dense, Reshape, Permute, Concat & Dot layers.

note: The keras layer allows running a loop on certain axes of the input tensor\s. In the current substitution axes that are looped on are folded to the bacth axis.

### CODE REVIEW

File: model_compression_toolkit/keras/graph_substitutions/substitutions/activation_decomposition.py [hunk 1]
```diff
@@ -61,6 +61,9 @@ def substitute(self,
         Returns:
             Graph after applying the substitution.
         """
+        # if op2d_node.framework_attr.get(ACTIVATION) in [None, LINEAR]:
+        #     # skip substitution if there's no activation
+        #     return graph
```
Comment (haihabi) [inline model_compression_toolkit/keras/graph_substitutions/substitutions/activation_decomposition.py:66]: Remove this comment from PR

File: model_compression_toolkit/keras/default_framework_info.py [hunk 2]
```diff
@@ -129,6 +131,7 @@
 ACTIVATION2MINMAX = {SOFTMAX: (0, 1),
                      SIGMOID: (0, 1),
                      LINEAR: (None, None),
+                     None: (None, None),
```
Comment (haihabi) [inline model_compression_toolkit/keras/default_framework_info.py:134]: Find the case when this is needed and let's talk about the solution

File: model_compression_toolkit/keras/graph_substitutions/substitutions/multi_head_attention_decomposition.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+            mha_node: MultiHeadAttention node to replace.
+
+        Returns:
+            Graph after applying the substitution.
+        """
+
+        query_nodes, key_nodes, value_nodes, att_head_output_nodes = [], [], [], []
+
+        # MHA params:
+        num_heads = mha_node.framework_attr['num_heads']
```
Comment (haihabi) [inline model_compression_toolkit/keras/graph_substitutions/substitutions/multi_head_attention_decomposition.py:62]: Move all string to constant

File: model_compression_toolkit/keras/graph_substitutions/substitutions/multi_head_attention_decomposition.py [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+        value_dim = mha_node.framework_attr['value_dim']
+        query_shape = tuple(mha_node.framework_attr['query_shape'])
+        key_shape = tuple(mha_node.framework_attr['key_shape'])
+        value_shape = tuple(mha_node.framework_attr['value_shape'])
+        output_dim = mha_node.framework_attr['output_shape']
+        d_model = query_shape[-1] if output_dim is None else output_dim
+        attention_axes = tuple(mha_node.framework_attr['attention_axes'])
+
+        iter_axes, iter_axes_prod, q_att_axes_prod, kv_att_axes_prod = [], 1, 1, 1
+        for i, (aq, ak) in enumerate(zip(query_shape[1:-1], key_shape[1:-1])):
```
Comment (haihabi) [inline model_compression_toolkit/keras/graph_substitutions/substitutions/multi_head_attention_decomposition.py:74]: Make here  functions with documentations
